pramen-py/src/pramen_py/metastore/reader.py

```diff
@@ -119,50 +196,58 @@ def get_latest_available_date(
+                file_pattern=f"{metastore_table.info_date_settings.column}=*",
+            )
+            self.validate_partitions_paths(files_paths, metastore_table)
+            dates_list = []
+            for path in files_paths:
+                dates_list.append(
+                    self._extract_date_from_path(
+                        path,
+                        metastore_table.info_date_settings.column,
+                        metastore_table.info_date_settings.format,
+                    )
+                )
+        else:
+            df_select = self._read_table(
+                metastore_table.format, metastore_table.path
+            ).select(f"{metastore_table.info_date_settings.column}")
+            dates_list = [row[0] for row in df_select.distinct().collect()]
 
-        def extract_dates(file_name: str) -> datetime.date:
-            return convert_str_to_date(
-                file_name.split("=")[1],
-                fmt=table.info_date_settings.format,
+        if not dates_list:
+            logger.error(
+                f"The directory does not contain partitions by "
+                f"'{metastore_table.info_date_settings.column}': {metastore_table.path}"
             )
+            raise ValueError("No partitions are available")
 
-        files = self.fs_utils.list_files(
-            table.path,
-            file_pattern=f"{table.info_date_settings.column}=*",
-        )
-        logger.debug(
-            f"The following files are in the {table.path}:\n"
-            f"{chr(10).join(files)}"
+        filtered_dates_list = self._filter_info_dates_by_until(
+            dates_list, until  # type: ignore
         )
 
-        try:
-            latest_date = max(
-                filter(
-                    before_until,
-                    map(
-                        extract_dates,
-                        files,
+        if not filtered_dates_list:
+            str_date_list = list(
+                map(
+                    lambda date: convert_date_to_str(
+                        date, metastore_table.info_date_settings.format
                     ),
+                    filtered_dates_list,
                 )
             )
-        except ValueError as err:
-            raise ValueError(
-                f"No partitions are available for the given {table_name}. "
-                f"The content of a table path is:\n"
-                f"{chr(10).join(files)}\n"
+            logger.error(
+                f"No partitions are available for the given '{metastore_table.name}'.\n"
+                f"The table is available for the following dates:\n"
+                f"{str_date_list}\n"
                 f"Only partitions earlier than {str(until)} might be included."
-            ) from err
-        else:
-            logger.info(f"Latest date for {table_name} is {latest_date}")
-            return latest_date
+            )
+            raise ValueError("No partitions are available")
```

Comment:
Same here - you can raise an error only, but it can contain the more detailed message.